File: csharp/src/Blockchain.Abstractions/Workflows/IEventListenerWorkflow.cs
using Temporalio.Workflows;

namespace Train.Solver.Blockchain.Abstractions.Workflows;

[Workflow]
public interface IEventListenerWorkflow
{
    [WorkflowQuery]
    ulong? GetLastScannedBlock();

    [WorkflowQuery]
    HashSet<string> ProcessedTransactionHashes();

    [WorkflowRun]
    Task RunAsync(string networkName, uint blockBatchSize, int waitInterval, ulong? lastProcessedBlockNumber = null);
}

File: js/src/Blockchian/Blockchain.Abstraction/Interfaces/IEventListenerWorkflow.ts
import { NetworkType } from '../../../Data/Entities/Networks';

export interface IEventListenerWorkflow {
  GetLastScannedBlock(): Promise<number | undefined>;

  ProcessedTransactionHashes(): Promise<Set<string>>;

  RunAsync(
    networkName: string,
    networkType: NetworkType,
    blockBatchSize: number,
    waitInterval: string,
    lastProcessedBlockNumber?: number
  ): Promise<void>;
}
  